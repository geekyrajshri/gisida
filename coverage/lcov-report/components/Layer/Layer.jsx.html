<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Layer/Layer.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Layer</a> Layer.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15</td><td class="line-coverage quiet"><span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';<span class="branch-0 cbranch-no" title="branch not covered" ></span>
require('./Layer.scss');
&nbsp;
const Layer = ({ mapTargetId, layer, layerData, visible = false, onLayerChange = f =&gt; f }) =&gt;
  (&lt;li className={`layer ${mapTargetId}`}&gt;
    &lt;input
      type="checkbox"
      data-layer={layer}
      onChange={e =&gt; onLayerChange(layer, e.target.checked, mapTargetId)}
    /&gt;
    &lt;label&gt;{layerData[layer].label}&lt;/label&gt;
  &lt;/li&gt;);
&nbsp;
export default Layer;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jun 27 2017 15:17:21 GMT+0300 (EAT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
