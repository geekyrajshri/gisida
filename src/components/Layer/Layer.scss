li {
  label {
    margin-left: 2%;
  }
}
