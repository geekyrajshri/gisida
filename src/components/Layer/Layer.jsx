import React from 'react';
import PropTypes from 'prop-types';

require('./Layer.scss');

function getType(defaultView) {
  const type = defaultView === 'framework' ? 'radio' : 'checkbox';

  return type;
}
const Layer = ({
  mapTargetId,
  layer,
  sector,
  layerData,
  headers, defaultView, onLayerChange = f => f }) =>
  (<li className={`layer ${mapTargetId}`}>
    {headers.includes(layer) ? <b>{layer}</b> :
    <label className="label" htmlFor={layer} >
      <input
        type={defaultView === 'framework' ? 'radio' : 'checkbox'}
        id={layer.replace(' ', '-')}
        data-layer={layer}
        name={sector}
        value={layer}
        onClick={e => onLayerChange(layer, e.target.checked, mapTargetId, sector,
        getType(defaultView))}
      />
      {layerData[layer].label}</label>
    }
  </li>);

Layer.propTypes = {
  mapTargetId: PropTypes.string.isRequired,
  layer: PropTypes.string.isRequired,
  layerData: PropTypes.objectOf(PropTypes.any).isRequired,
  onLayerChange: PropTypes.func.isRequired,
  headers: PropTypes.arrayOf(PropTypes.any),
  defaultView: PropTypes.string,
  sector: PropTypes.string,
};

Layer.defaultProps = {
  headers: [],
  sector: '',
  defaultView: '',
};

export default Layer;
