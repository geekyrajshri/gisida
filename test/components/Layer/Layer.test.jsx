import React from 'react';
import renderer from 'react-test-renderer';
import Layer from '../../../src/components/Layer/Layer';

describe('Layer', () => {
    it('Layer renders each layer indicator', () => {
        const component = renderer.create(<Layer />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});