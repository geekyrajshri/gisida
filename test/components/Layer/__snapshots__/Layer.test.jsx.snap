// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layer Layer renders each layer indicator 1`] = `
<li
  className="layer map-1"
>
  <label
    className="label"
    htmlFor="test_layer"
  >
    <input
      data-layer="test_layer"
      id="test_layer"
      name=""
      onClick={[Function]}
      type="checkbox"
      value="test_layer"
    />
    Test Layer
  </label>
</li>
`;
